### ISSUE TITLE
DatePicker Issue - Unable to select the dates for the second time

### ISSUE THREAD

**Issue by arun-prasad-r** [mydea/ember-date-components#198]
The DatePicker component appears to be broken. I am unable to select the date for the second time. 

```
"ember-date-components": "^4.0.0",
"ember-source": "3.24.3",
"ember-cli": "~3.24.0",
```

**Browser:** Chrome v90.*
**OS:** Mac BigSur

I can see the same issue in the demo page: 
http://mydea.github.io/ember-date-components/

### Steps to reproduce:
1. Select a date in any datepicker in the demo page. 
2. Now try to change the date by reselecting.

**Comment by mydea** [comment]
Huh, that appears to be bad! I will look into it. Need to add a test case for this, def. seems as if such a thing should be catched by the tests :grimacing:

**Comment by mydea** [comment]
Should be fixed in 4.0.1 - let me know if you encounter any other/further issues! Thanks for reporting this.

**Comment by arun-prasad-r** [comment]
@mydea I can confirm that this issue is fixed in 4.0.1. Thanks!
